starter_code/components.js

```diff
@@ -0,0 +1,31 @@
+
+
+class Image extends React.Component {
+  constructor(){
+    super();
+    this.state = {showResults: true}
+  }
+  _toggleImage(event){
+    this.setState( {showResults: !this.state.showResults} );
+  }
+  render() {
+    if (this.state.showResults){
```

Comment:
Rather than using a conditional here to render 2 different templates under different circumstances, is there a way we could alter the code to render just 1 template, but slightly differently depending on whether `this.state.showResults` was true or false? This would keep our code DRYer and easier to read and maintain.

Maybe something like:

```js
render() {
  return (
    <div>
      { this.state.showResults ? <img src="http://www.pamperedpetz.net/wp-content/uploads/2015/09/Puppy1.jpg"  /> : null }
      <button type="button" class="btn btn-primary" onClick={this._toggleImage.bind(this)} >Show/Hide</button>
    </div>
  )
}
```

This is a good opportunity to use the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) in your code to do a quick, inline conditional.